### PR TITLE
ECDSA library for signature recovery will revert when failling

### PR DESCRIPTION
**Pull request by cducrest**
It was previously returning the zero address instead of
reverting when the signature was invalid

related to: https://github.com/trustlines-network/project/issues/829

### CODE REVIEW

**Comment by cducrest** [comment]
How can it be that I had only one test failing on circleci and you updated more than one value and now we still have the same test failing.

**Comment by cducrest** [comment]
I guess a solution would be to make fixtures related to gas tests not session scoped.

**Comment by berndbohmeier** [comment]
Ah, yes I remembered that there was an issue that I was to lazy to fix, because it was nothing I touched, but because the tests were not properly isolated.

File: tests/conftest.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def assert_gas_values_match_for_call(
                     gas_limit=self.data[key].limit,
                 )
 
-        def assert_gas_costs_match(self, key, gas_cost, *, abs_delta=0):
+        def assert_gas_costs_match(self, key, gas_cost, *, abs_delta=100):
```
Comment (berndbohmeier) [inline tests/conftest.py:244]: I would strongly advice against setting the delta to 100. 
The gas costs should usually be predictable afaik, so the only reason I am aware of are flaky tests, So I would rather set this in the flaky tests.

Comment (cducrest) [reply tests/conftest.py:244]: I could not find a reason for the variation of gas usage from the tests and their fixtures, so I would not say so.

**Pull request merged**